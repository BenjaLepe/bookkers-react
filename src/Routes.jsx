import React from 'react';
import { Route, Switch } from 'react-router-dom';
import NotFound from './views/NotFound';
import Login from './views/Login';
import BookDetail from './views/BookDetail';
import SignUp from './views/SignUp';
import Home from './views/Home';
import AuthContextProvider from './contexts/AuthContext';
import SessionProfile from './views/SessionProfile';
import SessionExpired from './views/SessionExpired';
import UserProfile from './views/UserProfile';
import CreateBook from './views/AddBook';
import EditUser from './views/EditUser';
import Navbar from './Components/Navbar';
import reviewReports from './views/Reports';
import EditBook from './views/EditBook';

export default function Routes() {
  return (
    <AuthContextProvider>
      <Navbar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/sign-up" component={SignUp} />
        <Route exact path="/session-expired" component={SessionExpired} />
        <Route exact path="/session-profile" component={SessionProfile} />
        <Route exact path="/users/:id" component={UserProfile} />
        <Route exact path="/books/new" component={CreateBook} />
        <Route exact path="/books/:id" component={BookDetail} />
        <Route exact path="/edit-book/:id" component={EditBook} />
        <Route exact path="/edit-user/:id" component={EditUser} />
        <Route exact path="/books/:book_id/reviews/:review_id/reports" component={reviewReports} />
        <Route component={NotFound} />
      </Switch>
    </AuthContextProvider>
  );
}