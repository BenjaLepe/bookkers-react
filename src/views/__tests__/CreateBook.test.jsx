import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../../Routes';

const { create } = renderer;

function TestRouter({ path }) {

  return (
    <MemoryRouter initialEntries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

describe('CreateBook', () => {
  describe('When there is no current session', () => {
    it ('renders the home page.', () => {
      const DOM = create(<TestRouter path ="/books/new" />).toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });

  describe('When there is a current session', () => {
    const user = {
      access_token: 'TestToken',
      token_type: 'Bearer',
      user: {
        username: 'testUserName',
        first_name: 'testFirstName',
        last_name: 'testLastName',
        email: 'test@email.cl',
        description: 'testDescription',
      },
    };
    const sessionExp = new Date(
      new Date().getTime() + 1000 * 60 * 60 * 24, // One day in the future
    );
    const localStorageData = {
      user,
      sessionExp,
    }

    beforeEach(() => {
      global.Storage.prototype.getItem = jest.fn(
        (key) => JSON.stringify(localStorageData[key]),
      );
    });

    afterEach(() => {
      global.Storage.prototype.getItem.mockReset();
    });

    it('renders the create page form', () => {
      const DOM = create(<TestRouter path="/books/new" />).toJSON();
      expect(DOM).toMatchSnapshot();
    });
  });
});