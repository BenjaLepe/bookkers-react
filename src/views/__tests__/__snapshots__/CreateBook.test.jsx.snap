// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBook When there is a current session renders the create page form 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/session-profile"
          onClick={[Function]}
        >
          Profile
        </a>
        <a
          aria-current="page"
          className="nav-link active"
          href="/books/new"
          onClick={[Function]}
          style={Object {}}
        >
          Add book
        </a>
        <button
          className="nav-button"
          onClick={[Function]}
          type="button"
        >
          Log out
        </button>
      </div>
    </div>
  </div>,
  <div>
    <form
      action="#"
      className="form-box"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <h2>
        Add new book
      </h2>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="ISBN"
          >
            ISBN:
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="ISBN"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="editorial"
          >
            Editorial:
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="editorial"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="pages_number"
          >
            Pages number:
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="pages_number"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="author"
          >
            Author:
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="author"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="genre"
          >
            Genre:
          </label>
        </div>
        <div
          className="field-container"
        >
          <select
            name="genre"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              label="Select a genre"
              value=""
            />
            <option
              label="Fantasy"
              value="Fantasy"
            />
            <option
              label="Adventure"
              value="Adventure"
            />
            <option
              label="Romance"
              value="Romance"
            />
            <option
              label="Contemporary"
              value="Contemporary"
            />
            <option
              label="Dystopian"
              value="Dystopian"
            />
            <option
              label="Mystery"
              value="Mystery"
            />
            <option
              label="Horror"
              value="Horror"
            />
            <option
              label="Thriller"
              value="Thriller"
            />
            <option
              label="Paranormal"
              value="Paranormal"
            />
            <option
              label="Science Fiction"
              value="Science Fiction"
            />
            <option
              label="Memoir"
              value="Memoir"
            />
            <option
              label="Cooking"
              value="Cooking"
            />
          </select>
        </div>
      </div>
      <div
        className="form-group"
      >
        <div
          className="label-container"
        >
          <label
            htmlFor="image"
          >
            Image (url):
          </label>
        </div>
        <div
          className="field-container"
        >
          <input
            name="image"
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            value=""
          />
        </div>
        <div
          className="img-preview"
        />
      </div>
      <div
        className="submit-container"
      >
        <button
          type="submit"
        >
          Create Book
        </button>
      </div>
    </form>
    <p>
      
    </p>
  </div>,
]
`;

exports[`CreateBook When there is no current session renders the home page. 1`] = `
Array [
  <div
    className="nav-bg"
  >
    <div
      className="nav-bar container"
    >
      <div
        className="nav-group-logo"
      >
        <a
          aria-current="page"
          className="nav-link nav-title active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Bookkers
        </a>
      </div>
      <div
        className="nav-group"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/sign-up"
          onClick={[Function]}
        >
          Sign Up 
        </a>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="head-container"
    >
      <div
        className="logo-container"
      >
        <img
          alt="LOGO"
          className="logo"
          src="logo.png"
        />
      </div>
      <div
        className="welcome-container"
      >
        <h2>
          Welcome to Bookers
        </h2>
        <p>
          Bookkers es una red social que permite a sus usuarios publicar, calificar, recomendar y ver las recomendaciones que otros usuarios han hecho respecto a los libros ingresados a la aplicaci√≥n.
        </p>
      </div>
    </div>
  </div>,
]
`;
